File: mushroomAPI/mushroomAPI/Entities/Mushroom.cs
namespace mushroomAPI.Entities
{
   
    public class Mushroom
    {
        
        public int Id { get; set; }
        public string Name { get; set; } = "";
        public string ScientificName { get; set; } = "";
        public MushroomCategory Category { get; set; }
        public string Description { get; set; } = "";
        public bool IsEdible { get; set; }
        public string Habitat { get; set; } = "";
        public string Season { get; set; } = "";
        public List<string> CommonNames { get; set; } = new List<string>();
        public List<string> ImageUrls { get; set; } = new List<string>();
        public List<Coordinates> Locations { get; set; } = new List<Coordinates>();
        public DateTime LastUpdated { get; set; }
    
    }
}

File: webClient/src/contracts/mushroom/mushroom.ts
export enum MushroomCategory {
  Agaricus = 0,
  Amanita = 1,
  Boletus = 2,
  Cortinarius = 3,
  Entoloma = 4,
  Hygrocybe = 5,
  Lactarius = 6,
  Russula = 7,
  Suillus = 8
}

export interface Coordinates {
  id: number;
  latitude: number;
  longitude: number;
  userId: number;
  username: string;
}

export interface SaveCoordinatesDTO {
  latitude: number;
  longitude: number;
}

export interface MushroomPrediction {
  category: string;
  confidence: string;
}

export interface CategoriesDTO {
  predictions: MushroomPrediction[];
}

export interface MushroomDTO {
  id: number;
  name: string;
  scientificName: string;
  category: MushroomCategory;
  description: string;
  isEdible: boolean;
  habitat: string;
  season: string;
  commonNames: string[];
  imageUrls: string[];
  locations: Coordinates[];
  lastUpdated: Date;
}

export interface UpsertMushroomDTO {
  name: string;
  scientificName: string;
  category: MushroomCategory;
  description: string;
  isEdible: boolean;
  habitat: string;
  season: string;
  commonNames: string[];
  imageUrls: string[];
  locations: Coordinates[];
  lastUpdated: string;
}
